webapp/security/views.py

```diff
@@ -352,171 +352,148 @@ def cve(cve_id):
-                    source=package_data["source"],
-                    ubuntu=package_data["ubuntu"],
-                    debian=package_data["debian"],
-                    releases=releases,
-                )
-            )
 
-    if data.get("references"):
-        for ref in data["references"]:
-            references.append(CVEReference(uri=ref))
-
-    if data.get("bugs"):
-        for b in data["bugs"]:
-            bugs.append(Bug(uri=b))
-
-    cve = CVE(
-        id=data["id"],
-        status=data.get("status", ""),
-        last_updated_date=parse(data.get("last_updated_date"))
-        if data.get("last_updated_date")
-        else None,
-        public_date_usn=parse(data.get("last_updated_date"))
-        if data.get("public_date_usn")
-        else None,
-        public_date=parse(data.get("public_date"))
-        if data.get("public_date")
-        else None,
-        priority=data.get("priority"),
-        cvss=data.get("cvss"),
-        assigned_to=data.get("assigned_to"),
-        discovered_by=data.get("discovered_by"),
-        approved_by=data.get("approved_by"),
-        description=data.get("description"),
-        ubuntu_description=data.get("ubuntu_description"),
-        notes=data.get("notes", []),
-        packages=packages,
-        references=references,
-        bugs=bugs,
-    )
+        if data.get("references"):
+            for ref in data["references"]:
+                references.append(CVEReference(uri=ref))
+
+        if data.get("bugs"):
+            for b in data["bugs"]:
+                bugs.append(Bug(uri=b))
+
+        cve = CVE(
+            id=data["id"],
+            status=data.get("status", ""),
+            last_updated_date=parse(data.get("last_updated_date"))
+            if data.get("last_updated_date")
+            else None,
+            public_date_usn=parse(data.get("public_date_usn"))
+            if data.get("public_date_usn")
+            else None,
+            public_date=parse(data.get("public_date"))
+            if data.get("public_date")
+            else None,
+            priority=data.get("priority"),
+            cvss=data.get("cvss"),
+            assigned_to=data.get("assigned_to"),
+            discovered_by=data.get("discovered_by"),
+            approved_by=data.get("approved_by"),
+            description=data.get("description"),
+            ubuntu_description=data.get("ubuntu_description"),
+            notes=data.get("notes", []),
+            packages=packages,
+            references=references,
+            bugs=bugs,
+        )
+        cve_object.append(cve)
+    return cve_object
 
-    try:
-        db_session.add(cve)
-        db_session.commit()
-    # Duplicate key errors and Payload input errors
-    except (IntegrityError, DataError) as error:
-        return flask.jsonify({"message": error.orig.args[0]}), 409
 
-    return flask.jsonify({"message": "CVE created succesfully"}), 201
+def update_cve(cve_data):
```

Comment:
This should now be `update_cves` right? It expects to operate on a list. Same with `create_cves`. Although really it's more like `prepare_cves_to_update` or something, as it doesn't commit anything - using the language `update` or `create` makes it sound like it's going to actually do the action.

Please try to be careful with your singulars and plurals - they are logically extremely important and using the wrong one could easily lead to mistakes in development.